### PR TITLE
Merge delayer definitions and add protection checks

### LINKED ISSUE #488: Everyone can configure delayers in protected areas
the delay of delayers can be set by anyone in protected areas.
Issue found on BlockySurvival

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #488

### CODE REVIEW

File: mesecons_delayer/init.lua [hunk 1]
```diff
@@ -83,26 +64,46 @@ minetest.register_node("mesecons_delayer:delayer_off_"..tostring(i), {
...
+	wield_image = "mesecons_delayer_off_1.png",
+	groups = on_groups,
+	drop = "mesecons_delayer:delayer_off_1",
+	on_punch = function(pos, node, puncher)
+		if minetest.is_protected(pos, puncher and puncher:get_player_name()) then
+			return
 		end
+
+		minetest.swap_node(pos, {
+			name = "mesecons_delayer:delayer_off_"..tostring(next_i[i]),
```
Comment (DS-Minetest) [inline mesecons_delayer/init.lua]:
```suggestion
			name = "mesecons_delayer:delayer_off_"..tostring(i % 4 + 1),
```

**Comment by DS-Minetest** [comment]
Would it be possible to use `mesecon.register_node`?

**Comment by SmallJoker** [comment]
Maybe, but I neither know the difference nor am I familiar with the mesecons API.

**Comment by SmallJoker** [comment]
@numberZero Would you please be so nice and review this PR if you've got some free time?

**Pull request merged**